### ISSUE TITLE
svc(): wire the continuous NNGP spatially-varying coefficient to the Laplace backends for occu()

### ISSUE THREAD

**Issue by gcol33** [gcol33/tulpaObs#143]
## Current state

`svc()` is consumed only by the single-season occupancy NUTS path (`populate_svc`,
`src/populate_helpers.h` -> `src/occu_fit.cpp`). On `occu()` under `laplace` /
`nested_laplace` the guard in `.tobs_fit_model()` (R/occu_fit.R) errors:

```
svc() (a spatially-varying coefficient) is only wired for single-season occu()
under method = "nuts" ...
```

That guard is correct (before #118 the term was silently dropped and the model fit
without it). But it means the package's default backend cannot fit the term at all: a
user who writes `svc()` has to switch to `method = "nuts"` for the whole model.

## Ask

Support `svc()` on `occu()` under `laplace` / `nested_laplace`.

The continuous analogue of the recovery-tested weighted areal bar
(`spatial(~ 1 + w || cell, graph)`, which arrives as a `spatial` term and already fits
under `nested_laplace`) is a covariate-weighted continuous field block. The areal SVC
already generalises to K covariate-weighted fields solved jointly
(`.tobs_areal_field_blocks`, `.areal_field_car(weight=)`), so the shape of the solve is
established; what is missing is the same weighting on a continuous (spde / NNGP) block
in the multi-block nested-Laplace prior.

## Checklist

- [ ] Decide the route: weighted `spde()` block in the multi-block prior (mesh nodes,
      barycentric projector already exists via `fit$spatial$tulpa_spec$A`) vs a weighted
      NNGP block matching `svc()`'s current NNGP parameterization
- [ ] Thread the covariate weight through the block so `eta_i += w_i * z(s_i)`
- [ ] Reconstruct and expose the surface (`fit$svc_field`, matching the NUTS path's
      naming) plus the range/SD hyperparameters
- [ ] Recovery test against a known simulated SVC surface, asserting the coefficient
      surface by correlation and the hyperparameters, in the style of
      `test-occu-spatial-svc-recovery.R`
- [ ] Relax the `.tobs_fit_model()` guard for the routes that then work, keeping the
      error (with a pointer) for the rest
- [ ] `test-svc-guard.R`: the `occu()` + `laplace` case currently asserts an error and
      will need updating

## Notes

The eta assembly, NNGP prior, and gradient for the NUTS path live in the compiled
upstream tulpa engine, not in tulpaObs, so this may need a tulpa-side counterpart.

Related: #118 (the guard), #119 (NUTS SVC recovery validation).

**Comment by gcol33** [comment]
Shipped in 3b9ce67.

`occu() + svc()` now fits under `method = "laplace"` / `"nested_laplace"`. The K surfaces are latent field blocks on the psi logit, so the fit rides the shared areal-BFGS nested-Laplace driver (`.tobs_areal_bfgs_fit`) rather than a new engine. Two new pieces only:

- `.tobs_svc_nngp_field()` — a continuous NNGP block with an optional per-site design weight, the continuous sibling of `.areal_field_car(weight = )`
- `.tobs_occu_svc_marginal()` — the exact two-state occupancy marginal with Fisher-identity gradients (`w - psi`, `w(y - p)`), FD-validated

**Route taken** (checklist item 1): a weighted NNGP block matching `svc()`'s own parameterization, not a weighted `spde()` block. The Vecchia precision `Q = (I - A)' D^-1 (I - A)` is assembled in R from the term's own neighbour structure, using the compiled kernel's kernels, jitter and conditional-variance floor. That is what makes it safe to have two implementations under one term name, and the test asserts it: the R log-density equals tulpa's `cpp_test_svc_nngp_twins" to 1e-8 across all three covariance kernels. No tulpa-side counterpart was needed.

Hyperparameters (sigma, phi) are grid-integrated on both routes, so `laplace` and `nested_laplace` land on the same fitter; reported as `fit$svc_hyper`, surface as `fit$svc_field` (NUTS naming). `fitted()` reads the surface in sample via `model$occ_eta_offset`; `predict(newdata=)` does not krige to new locations, as on NUTS.

**Recovery**: surface cor 0.78 / 0.60 / 0.83 on seeds 1/2/3 at N=150, J=6, p=0.6, against the NUTS path's 0.76 / 0.60 / 0.81 on the same truth. The surface is information-bounded at these settings and both backends sit at that bound, so the floor matches the NUTS test's (0.55) rather than being tuned to these seeds. A covariate-weighted surface is tested separately — with the weight ignored the block would fit a plain varying intercept and that correlation would collapse, so it is the direct evidence the design weight is fit.

**Guard narrowed, not removed**: a detection-arm svc, a non-single family, a spatial/temporal/re term alongside svc, and `pg_gibbs` all still error with a pointer rather than silently dropping the term.

`test-occu-svc-laplace-recovery.R` (36 assertions, 0 failures) + `test-svc-guard.R` (8, 0). Docs discoverability stays open as #145; the two-entry-point API question as #146.
